includes/functions.php

```diff
@@ -1300,12 +1300,24 @@ function get_content_warning( $post_id ) {
 /**
  * Get the ActivityPub ID of a User by the WordPress User ID.
  *
+ * Fall back to blog user if in blog mode or if user is not found.
+ *
  * @param int $id The WordPress User ID.
  *
- * @return string The ActivityPub ID (a URL) of the User.
+ * @return string|false The ActivityPub ID (a URL) of the User or false if not found.
  */
 function get_user_id( $id ) {
-	$user = Actors::get_by_id( $id );
+	$mode = \get_option( 'activitypub_actor_mode', 'default' );
+
+	if ( ACTIVITYPUB_BLOG_MODE === $mode ) {
+		$user = Actors::get_by_id( 0 );
+	} else {
+		$user = Actors::get_by_id( $id );
+	}
+
+	if ( \is_wp_error( $user ) ) {
+		$user = Actors::get_by_id( 0 );
```

Comment:
```suggestion
		$user = Actors::get_by_id( Actors::BLOG_USER_ID );
```
This wp_error check only seems relevant to the `else` conditional of `Actors::get_by_id( $id );`, let's move it in there? If the result of `Actors::get_by_id( Actors::BLOG_USER_ID );` is a wp_error, there's no reason to try again.